pkg/metrics/ingest.go

```diff
@@ -41,12 +43,28 @@ type MQTTPayload map[string]interface{}
 
 func (i *Ingest) store(deviceID string, rawMetrics MQTTPayload) error {
 	var mc MetricCollection
+
 	for metricName, value := range rawMetrics {
 		if cfg, found := i.validMetrics[metricName]; found {
+			var floatValue float64
+			var isFloat bool
+			var err error
+			floatValue, isFloat = value.(float64)
+			if !isFloat {
+				stringValue, isString := value.(string)
+
+				if !isString || ! validNumber.MatchString(stringValue) {
+					return fmt.Errorf("got data with unexpectd type: %T ('%s')", value, value)
+				}
 
-			floatValue, ok := value.(float64)
-			if !ok {
-				return fmt.Errorf("got data with unexpectd type: %T ('%s')  but wanted float64", value, value)
+				if validNumber.MatchString(stringValue) {
```

Comment:
We don't need this `if` anymore since the negated condition is already checked in line 56. If we are this far, `stringValue` has matched the regex.